packages/app/src/ChartUtils.tsx

```diff
@@ -820,3 +835,196 @@ export const convertV1ChartConfigToV2 = (
   }
   throw new Error(`unsupported table in v2: ${firstSeries.table}`);
 };
+
+/**
+ * Build search URL for viewing events based on group-by values
+ * Used by both chart clicks and table row clicks
+ */
+export function buildEventsSearchUrl({
+  source,
+  config,
+  dateRange,
+  groupFilters,
+  valueRangeFilter,
+}: {
+  source: TSource;
+  config: ChartConfigWithDateRange;
+  dateRange: [Date, Date];
+  groupFilters?: Array<{ column: string; value: any }>;
+  valueRangeFilter?: { expression: string; value: number; threshold?: number };
+}): string | null {
+  if (!source?.id) {
+    return null;
+  }
+
+  const isMetricChart = isMetricChartConfig(config);
+  if (isMetricChart && source?.logSourceId == null) {
+    notifications.show({
+      color: 'yellow',
+      message: 'No log source is associated with the selected metric source.',
+    });
+    return null;
+  }
```

Comment:
I'm wondering, should we allow any deeplinking for metric charts? There are columns in metric tables which can be in the `where` which would result in there being an error when arriving on the search page.

If we do want to link metrics, should we just do so by time range and not include the filters/where?

Reply:
The user may want to link to traces insetad of logs, can do that as a follow up. Additionally, we can try to filter by resource attributes (if any).